Michael Brady - DS3 - Unit 2 - Sprint 3

mbrady4 opened the pull request.

mbrady4 committed:
Lecture updates

mbrady4 committed:
Updates to Assignment

mbrady4 committed:
Updates to assignment

mbrady4 committed:
Added .gitignore

mbrady4 committed:
Updates

mbrady4 committed:
Refactored Model Resulting in significant accuracy gain

- Grouped infrequent genres into an 'other' category
- Used PCA to reduce dimensionality of audio features
- Used pipelines and column transformer to simplify code

veritaem reviewed (commented):
Did you redo the lesson code for assignment 2 or sis I just grab the wrong commit?  Either way good job on your refactored model, 50% accuracy on a multiclass regression with this kind of dirty data isnt bad at all, and I very much like the hash pipeline!

mbrady4 committed:
Uploaded Side Work w/Appts Dataset

After replicating lecture notebook from yesterday, I refactored work on a medical appointments dataset I have been working with for fun.

Hoping to evantually get above the majority classifier!!!

mbrady4 committed:
Created Data Dict for Bank Data

mbrady4 committed:
General Updates

mbrady4 committed:
Create Notebook to Predict High Potential Sales Targets

Slightly beating a majority classifier, but given the imbalanced dataset it seems like a solid start

mbrady4 committed:
General Updates

mbrady4 committed:
Built an interactive ROC Curve Visualization

Per Ryan's suggestion, using ipython widgets built a function to interactively model thresholds.

veritaem reviewed (commented):
It looks good! Hoever the predicted positive looks like its not being predicted enough given its actual distribution, is there any way you could take factors correlated with a 'yes' and make them more BLUE?

mbrady4 commented:
> It looks good! Hoever the predicted positive looks like its not being predicted enough given its actual distribution, is there any way you could take factors correlated with a 'yes' and make them more BLUE?

What do you mean by BLUE? Today, I tried oversampling the 'yes' observations which helped somewhat balance out the predictions.

mbrady4 committed:
Lecture Notebooks

mbrady4 committed:
Updated Banking Sales Success Predictor

Added Decision Tree, RandomForest, and XGB classifier models

mbrady4 committed:
Sprint Challenge Complete (output may or may not be there)

mbrady4 committed:
Completed Sprint Challenge (Now with Full Output)
